### PR TITLE
More strict: deprecate key-handler

### PR DESCRIPTION
**Pull request by chrisnojima**
Instead of all this logic to import or not the key handler (as a hoc nonetheless) this makes common-adapters/hot-key which exposes a hook and a simple component (that does nothing on mobile)

### CODE REVIEW

File: shared/chat/conversation/attachment-fullscreen/index.desktop.tsx [hunk 1]
```diff
@@ -1,53 +1,48 @@
...
 
   render() {
     return (
-      <PopupDialog onClose={this.props.onClose} fill={true}>
-        <Box style={styles.container}>
-          <Box style={styles.headerFooter}>
-            <Text type="BodySemibold" style={{color: Styles.globalColors.black, flex: 1}}>
+      <Kb.PopupDialog onClose={this.props.onClose} fill={true}>
+        <Kb.Box style={styles.container}>
+          <Kb.HotKey hotKeys={this.hotKeys} onHotKey={this.onHotKey} />
```
Comment (jakob223) [inline shared/chat/conversation/attachment-fullscreen/index.desktop.tsx]: feels weird to me to call this component `HotKey` rather than `HotKeyHandler` or something. Is this a pattern we use elsewhere?

File: shared/chat/inbox/row/chat-filter-row/index.tsx [hunk 2]
```diff
@@ -51,6 +52,13 @@ class ConversationFilterInput extends React.PureComponent<Props> {
     }
   }
 
+  private hotKeys = Platforms.isDarwin ? ['command+n'] : ['ctrl+n']
+  private onHotKeys = (cmd: string) => {
+    if (cmd.endsWith('+n')) {
```
Comment (jakob223) [inline shared/chat/inbox/row/chat-filter-row/index.tsx]: why is this necessary if the only hotkey ends with +n? What else would get passed here?

File: shared/common-adapters/hot-key.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        e.stopPropagation()
+        key && cb(key)
+      },
+      'keydown'
+    )
+    return () => {
+      unbind(keys)
+    }
+    // eslint-disable-next-line
+  }, []) // on mount only
```
Comment (jakob223) [inline shared/common-adapters/hot-key.tsx]: Don't want a component to be able to change its hotkeys by changing what keys it passes to this hook?

Comment (chrisnojima) [reply shared/common-adapters/hot-key.tsx]: this is more or less a port of the previous version. we never dynamically mutate the keys so i'm keeping it simple for now
